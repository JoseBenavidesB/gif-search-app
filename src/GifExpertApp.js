import React, { useState } from 'react'
import { AddCategory } from './components/AddCategory';
import { GifGrid } from './components/GifGrid';


//import PropTypes from 'prop-types'

function GifExpertApp( {defaultCategories = []} ) {
    
    //const categories = ['One punch', 'One Piece', 'Dragon Ball']
    const [categories, setCategories] = useState(defaultCategories);

   /*  const handleAdd = ()=>{
        //setCategories( ['Hunter X Hunter', ...categories ]  );
        setCategories( cats => ['Hunter X Hunter', ...cats] );
    }; */


    return (
        <>
            <h2>GifExpertApp (Find your gif)</h2>
            < AddCategory setCategories= {setCategories} />
            <hr/>


            <ol>
                {
                    categories.map( category => (
                         <GifGrid 
                            key={category}
                            category = {category}/>
                    ))
                }
            </ol>
        </>
    )
}

//GifExpertApp.propTypes = {}

export default GifExpertApp
