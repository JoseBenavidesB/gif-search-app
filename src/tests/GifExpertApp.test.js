import React from 'react';
import { shallow } from 'enzyme';

import GifExpertApp from '../GifExpertApp';


describe('Test <GifExpertApp/>', () => { 

    test('should return <GifExpertApp/> correctly', () => {

        const wrapper = shallow(<GifExpertApp/>)

        expect( wrapper ).toMatchSnapshot();
    });

    test('should a category list', () => {

        const categories = ['Dragon Ball', 'Goku'];
        const wrapper = shallow( <GifExpertApp defaultCategories={ categories }/>);

        expect( wrapper ).toMatchSnapshot();
        expect( wrapper.find('GifGrid').length ).toBe( categories.length );

    });
});
