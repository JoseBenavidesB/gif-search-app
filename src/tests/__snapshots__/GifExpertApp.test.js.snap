// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test <GifExpertApp/> should a category list 1`] = `
<Fragment>
  <h2>
    GifExpertApp (Find your gif)
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="Dragon Ball"
      key="Dragon Ball"
    />
    <GifGrid
      category="Goku"
      key="Goku"
    />
  </ol>
</Fragment>
`;

exports[`Test <GifExpertApp/> should return <GifExpertApp/> correctly 1`] = `
<Fragment>
  <h2>
    GifExpertApp (Find your gif)
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;
